src/cluster.c

```diff
@@ -1937,12 +2004,36 @@ int clusterStartHandshake(char *ip, int port, int cport) {
      * handshake. */
     n = createClusterNode(NULL,CLUSTER_NODE_HANDSHAKE|CLUSTER_NODE_MEET);
     memcpy(n->ip,norm_ip,sizeof(n->ip));
-    n->port = port;
+    if (server.tls_cluster) {
+        n->tls_port = port;
+    } else {
+        n->tcp_port = port;
+    }
     n->cport = cport;
     clusterAddNode(n);
     return 1;
 }
 
+static void getClientPortFromClusterMsg(clusterMsg *hdr, int *tls_port, int *tcp_port) {
+    if ((hdr->mflags[0] & CLUSTERMSG_FLAG0_FIXED_PORT) || server.tls_cluster) {
```

Comment:
```suggestion
    if (server.tls_cluster) {
```

I think we have to just rely on this config for now. We know that this is an immutable parameter.